### ISSUE TITLE
Empty file with suffix '.None' created

### ISSUE THREAD

**Issue by synox** [RhetTbull/osxphotos#353]
![Screenshot 2021-01-22 at 09 04 51](https://user-images.githubusercontent.com/2250964/105463829-e8759c80-5c90-11eb-8b67-9b1c6820bac1.png)

The file is 'missing' in the library would have to be downloaded from icloud. 

```
Exporting 20814 photos to /Users/aravindo/Pictures/Apple-Photos-Exported...
Processed: 20814 photos, exported: 0, updated: 0, skipped: 22629, updated EXIF data: 0, missing: 25, error: 0
Elapsed time: 20.098 seconds
```
and stderr: 
```
Error exporting photo, file /Users/aravindo/Pictures/Apple-Photos-Exported/2021-01-20/IMG_1411.PNG: a bytes-like object is required, not 'NoneType'
```

Now I have downloaded the image in Photos.app and now the export worked. 
![Screenshot 2021-01-22 at 09 08 49](https://user-images.githubusercontent.com/2250964/105464209-7487c400-5c91-11eb-9368-5cceb6162fb4.png)

Let me know if you need more information. Feel free to mark it as resolved.

**Comment by RhetTbull** [comment]
Please post the entire command line you used for export.  

I'll do some digging to see where this happened.  Unfortunately in a recent release I caught all the exceptions that occurred in an effort to make the error messages more "user friendly" and not have osxphotos crash on experiencing errors---but that means I don't have the normal context, including line number, from an error message!

**Comment by RhetTbull** [comment]
Also, I noticed the summary line says "error: 0" but it did print out this error so that's also a bug -- the code should have counted this error for the summary.

**Comment by RhetTbull** [comment]
I've been able to re-produce this and figured out where in the code it's happening.  If I use --use-photokit with an image that's missing but turn off the wifi so it can't download, the PhotoKit code creates a valid Photo object but then tries to write empty bytes:

```pycon
>>> import osxphotos
photolib = >>> photolib = osxphotos.photokit.PhotoLibrary()
>>> photo = photolib.fetch_uuid("F2479013-97CB-4A6C-90FE-806D470DDD5D")
>>> photo.export("/tmp","foo2")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/rhet/Dropbox/Code/osxphotos/osxphotos/photokit.py", line 503, in export
    fd.write(imagedata.image_data)
TypeError: a bytes-like object is required, not 'NoneType'
```

Will work on a fix this weekend.

**Comment by synox** [comment]
thank you. 

Your test setup makes sense. I often have poor WiFi, so this probably happened when I was not at the machine and the connection was broken. 

my command: 
```
# only need to trigger downloading on the first run. Excluding deleted, as they can not be downloaded and otherwise we get warnings. 
/Users/aravindo/.local/bin/osxphotos \
	export --update    --strip   --missing --download-missing  --use-photokit  \
	--directory "{created.date} {place.name.city,} {place.country_code,}" \
	--db "/Users/aravindo/Pictures/Photos Library.photoslibrary" \
	"/Users/aravindo/Pictures/Apple-Photos-Exported"

# export by date and location
/Users/aravindo/.local/bin/osxphotos \
	export --update  --strip    --deleted   \
	--directory "{created.date} {place.name.city,} {place.country_code,}" \
	--db "/Users/aravindo/Pictures/Photos Library.photoslibrary" \
	"/Users/aravindo/Pictures/Apple-Photos-Exported"
```

**Comment by RhetTbull** [comment]
I think this is fixed in [v0.40.2](https://github.com/RhetTbull/osxphotos/releases/tag/v0.40.2)  Let me know if you still experience this issue.

**Comment by synox** [comment]
Looks good, thank you!
